src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```
@@ -0,0 +1,117 @@
+                    + manufacturer.getName() + "; country = " + manufacturer.getCountry(), e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String selectQuery = "SELECT * FROM manufacturer WHERE id = ? AND is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement = connection
+                        .prepareStatement(selectQuery)) {
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            Manufacturer manufacturer = null;
+            if (resultSet.next()) {
+                manufacturer = parseManufacturer(resultSet);
+            }
+            return Optional.ofNullable(manufacturer);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer by id = " + id, e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        String selectQuery = "SELECT * FROM manufacturer WHERE is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllManufacturersStatement = connection
+                        .prepareStatement(selectQuery)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery();
+            while (resultSet.next()) {
+                Manufacturer manufacturer = parseManufacturer(resultSet);
+                allManufacturers.add(manufacturer);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all data from database", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        if (manufacturer == null) {
+            return null;
+        }
```

```suggestion
```
you should rather throw an exception if the manufacturer is null. But exceptions would occur while trying to get fields from the object. So this check is not needed here